bpf/go_nethttp.h

```diff
@@ -320,6 +322,8 @@ int beyla_uprobe_ServeHTTPReturns(struct pt_regs *ctx) {
     trace->type = EVENT_HTTP_REQUEST;
     trace->start_monotime_ns = invocation->start_monotime_ns;
     trace->end_monotime_ns = bpf_ktime_get_ns();
+    trace->host[0] = 0;
+    trace->scheme[0] = 0;
```

Comment:
```suggestion
    trace->method = '\0';
    trace->path = '\0';
    trace->host[0] = '\0';
    trace->scheme[0] = '\0';
```
- ensure method and path are always null-terminated as well (it matters in spanner.go I guess)
- just being pedantic here, so feel free to ignore - `\0` helps the reader visualise we are dealing with a string and not with some sort of array of `int`.

Reply:
ok, sure, I'll replace those.

Reply:
@rafaelroquetto can you explain a bit more that? how is `0` equal to `'\0'`? shouldn't be just empty string?

Reply:
@marctc of course!   - all of those above are C strings. A C string is defined to be an array of `char` that ends with the *null terminator*, i.e. the number `0`. 

So the string literal

```
const char rabbit_str[] = "Rabbit";
```

is equivalent to

```
const char rabbit_[7] = { 'R', 'a', 'b', 'b', 'i', 't', '\0' };
// or
const char rabbit_[7] = { 'R', 'a', 'b', 'b', 'i', 't', 0 };
```

The *null terminator* tells string manipulation functions where the end of the string is, for instance, to calculate the length of  a C string, we can do

```
size_t strlen(const char* str)
{
   size_t i = 0;
   while (*str++ != '\0') ++i;
   return i;
}
```

 In C, the `char` literal `'\0'` has the value of `0`. They are the same, and have the same size (usually `sizeof(int)`). In C++ they are not the same - `0` is an `int` literal whose size is `sizeof(int)`, whereas `'\0'` is a char literal whose size is `sizeof(char)` - so `sizeof(0) != sizeof('\0')` in C++.

What @grcevski is doing here is simply setting the first element of those char arrays to the null-terminator `\0`, which represents an empty C-string:

```
char something[16]; // uninitialised -> garbage
something[0] = '\0'; // something is now { '\0', ... garbage ... };
printf("str: '%s', size = %lu\n", something, strlen(something)); // prints "str: '' size = 0"
```

In other words, it suffices to set the first char as `0` instead of zeroing the entire buffer. My point was then to suggest using the char literal `'\0'` rather than the `int` literal `0` to explicitly indicate to the reader these buffers are actually carrying C strings.

Reply:
> In other words, it suffices to set the first char as `0` instead of zeroing the entire buffer. My point was then to suggest using the char literal `'\0'` rather than the `int` literal `0` to explicitly indicate to the reader these buffers are actually carrying C strings.

gotcha, thanks for the explanation!